src/components/layout/CyVerseAppBar.js

```diff
@@ -241,8 +241,19 @@ function CyverseAppBar(props) {
     const [bootstrapError, setBootstrapError] = useState(null);
     const [bootstrapQueryEnabled, setBootstrapQueryEnabled] = useState(false);
     const [profileRefetchInterval, setProfileRefetchInterval] = useState(null);
+
     const setPreferences = usePreferences()[1];
 
+    if (activeView === NavigationConstants.APPS) {
+        filter = searchConstants.APPS;
+    } else if (activeView === NavigationConstants.ANALYSES) {
+        filter = searchConstants.ANALYSES;
+    } else if (activeView === NavigationConstants.DATA) {
+        filter = searchConstants.DATA;
+    } else {
+        filter = searchConstants.ALL;
+    }
```

Comment:
So, I don't want to be a pest here, but I'm honestly a bit worried that this approach limits the usefulness of global search for our users, and I'm wondering if we can do better.

Specifically, doing it this way means that for the majority of pages in our app, a filter will already be applied; discovering that the search can search across everything at once requires either:

a.) noticing the filter dropdown, using it, noticing that there's an "All" option, and trying it. Some folks will do this, but more novice users, who are often our target, aren't that likely to get past the first or second step.
b.) trying the search on one of the small number of pages that doesn't have a filter in place. The dashboard is the most likely one of these, but it still seems like a small funnel for people to know about the feature.

All that being said, I think it's obviously valuable to prioritize searching within the kind of thing you're already looking at. I don't think this needs to be changed immediately or anything, but I wonder if we should add another pathway to the global search, perhaps another entry in the results similar to the "view all <data/apps/analyses> results for <search term>" that instead points users towards the global search, maybe something like (in the data view) "View Apps and Analyses for <search term>" (or similarly for the other filters). This would be especially valuable when there's no results for the given filter, so a user can with one click go from a "no results" search within one subtype, to a broader global search that may give them the results they want.

Anyway, food for thought.

Reply:
I like this idea. I can easily implement it too. I will wait for few others to comment to get a broader feedback.

Reply:
Good points.  I think another thing could be if you're in the Apps view, for example, then the app search results could be on top as opposed to always having Data results on top.  That might also help prioritize without hiding stuff with a filter?  Either way, I do like the idea of adding another entry to the results that signals to users that it can be a global search 👍

Reply:
I like the idea of a `view all` or `search all` option when searching under a filter 👍

Reply:
@ash3rz - Only problem is right now we can't control which search (app / data / analyses) results comes back first. We can still do some thing to makes sure the results are properly ordered but that will take changes / refactor to get it right.

Reply:
I will try to implement @ianmcorvidae suggestion for now. Let see what our users think.

Reply:
Ok I have implemented @ianmcorvidae suggestion. Can I get a review again ? Thanks.